packages/dds/tree/src/test/utils.ts

```diff
@@ -38,14 +38,45 @@ import { brand, makeArray } from "../util";
 
 // Testing utilities
 
-export function deepFreeze<T>(object: T): void {
-    // Retrieve the property names defined on object
-    const propNames: (keyof T)[] = Object.getOwnPropertyNames(object) as (keyof T)[];
-    // Freeze properties before freezing self
-    for (const name of propNames) {
-        const value = object[name];
-        if (typeof value === "object") {
-            deepFreeze(value);
+const frozenMethod = () => {
+    assert.fail("Object is frozen");
+};
+
+/**
+ * Recursively freezes the given object.
+ * @param object - The object to freeze.
+ * @param freezeMethods - Whether to freeze mutation methods on `Set` and `Map` instances.
+ * Passing `true` mutates the `Set` and `Map` instances encountered in the recursion.
+ */
+export function deepFreeze<T>(object: T, freezeMethods: boolean = true): void {
+    if (object instanceof Map) {
+        for (const [key, value] of object.entries()) {
+            deepFreeze(key, freezeMethods);
+            deepFreeze(value, freezeMethods);
+        }
+        if (freezeMethods && !Object.isFrozen(object)) {
```

Comment:
If the map is already frozen, should we assert that its mutation methods are all equal to frozenMethod?